src/Ignitor/SignalsListener/index.ts

```diff
@@ -11,22 +11,36 @@ import { ApplicationContract } from '@ioc:Adonis/Core/Application'
 
 /**
  * Exposes the API to invoke a callback when `SIGTERM` or
- * `SIGINT (pm2 only)` signals are received.
+ * `SIGINT` signals are received.
  */
 export class SignalsListener {
   protected onCloseCallback?: () => Promise<void>
 
+  private isKilling = false
+
   /**
    * Invoke callback and exit process
    */
-  private kill = async function () {
-    try {
-      await this.onCloseCallback()
+  private kill = async () => {
+    if (!this.isKilling) {
+      // First attempt. Try to kill the process gracefully.
+      this.isKilling = true
+      if (process.stderr.isTTY) {
+        // If the process is running in a terminal, display a message to the user.
+        console.error('Gracefully shutting down the process... Press CTRL+C to force it')
+      }
+      try {
+        await this.onCloseCallback()
+        process.exit(0)
+      } catch (error) {
+        // TODO: log error?
+        process.exit(1)
```

Comment:
It seems useful to log an error that may occur during shutdown, but maybe it's already done before and that's why the function here doesn't do it?